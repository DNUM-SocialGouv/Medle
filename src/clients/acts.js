import { saveAs } from "file-saver"
import fetch from "isomorphic-unfetch"

import { ACTS_ENDPOINT, API_URL } from "../config"
import { handleAPIResponse } from "../utils/errors"
import { METHOD_DELETE, METHOD_POST, METHOD_PUT } from "../utils/http"

export const findAct = async ({ id, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT + "/" + id, { headers })
  return handleAPIResponse(response)
}

export const searchActsByKey = async ({ key, value, headers }) => {
  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}?${key}=${value}`, {
    headers: { ...headers, "Content-Type": "application/json" },
  })

  const { elements } = await handleAPIResponse(response)
  return elements
}

const buildQueryParams = (params) => {
  const { search, startDate, endDate, hospitals, profiles, asker, requestedPage } = params

  const arr = []
  if (search) {
    arr.push(`fuzzy=${search}`)
  }
  if (startDate) {
    arr.push(`startDate=${startDate}`)
  }
  if (endDate) {
    arr.push(`endDate=${endDate}`)
  }
  if (hospitals?.length) {
    arr.push(`hospitals=${hospitals.map((elt) => elt.value).join(",")}`)
  }
  if (profiles?.length) {
    arr.push(`profiles=${profiles.map((elt) => elt.value).join(",")}`)
  }
  if (asker?.value) {
    arr.push(`asker=${asker?.value}`)
  }
  if (requestedPage) {
    arr.push(`requestedPage=${requestedPage}`)
  }
  return arr.length ? "?" + arr.join("&") : ""
}

export const searchActs = async (params) => {
  const { headers } = params

  const queryParams = buildQueryParams(params)

  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}${queryParams}`, { headers })

  return handleAPIResponse(response)
}

export const createAct = async ({ act, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT, {
    method: METHOD_POST,
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify(act),
  })
  return handleAPIResponse(response)
}

export const updateAct = async ({ act, headers }) => {
  const response = await fetch(API_URL + ACTS_ENDPOINT + "/" + act.id, {
    method: METHOD_PUT,
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify(act),
  })
  return handleAPIResponse(response)
}

export const deleteAct = async ({ id, headers }) => {
  const response = await fetch(`${API_URL}${ACTS_ENDPOINT}/${id}`, { method: METHOD_DELETE, headers })
  return handleAPIResponse(response)
}

export const fetchExport = async (params) => {
  const { headers } = params

  const queryParams = buildQueryParams(params)

  saveAs(`${API_URL}${ACTS_ENDPOINT}/export${queryParams}`, { headers })
}
