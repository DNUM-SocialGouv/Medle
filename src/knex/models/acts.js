import moment from "moment"

export const buildActsFromDB = acts =>
   acts.map(elt => ({
      id: elt.id,
      createdAt: elt.created_at,
      updatedAt: elt.updated_at,
      deletedAt: elt.deleted_at,
      casesFk: elt.cases_fk,
      pvNumber: elt.pv_number,
      examinationDate: elt.examination_date,
      asker: elt.asker,
      addedBy: elt.added_by,
      data: elt.data,
   }))

// from js to db
export const buildActFromJSON = data => {
   const res = { extra_data: {} }

   const mainKeys = {
      id: "id",
      internalNumber: "internal_number",
      pvNumber: "pv_number",
      examinationDate: "examination_date",
      profile: "profile",
      askerId: "asker_id",
      addedBy: "added_by",
      hospitalId: "hospital_id",
   }

   Object.keys(data).forEach(key => {
      if (mainKeys[key]) {
         res[mainKeys[key]] = data[key]
      } else {
         res.extra_data[key] = data[key]
      }
   })

   return res
}

// from db to js
export const buildActFromDB = data => ({
   id: data.id,
   internalNumber: data.internal_number,
   pvNumber: data.pv_number,
   examinationDate: moment(data.examination_date).format("YYYY-MM-DD"),
   profile: data.profile,
   asker: { id: data.asker_id, name: data.asker_name },
   user: {
      id: data.added_by,
      firstName: data.user_first_name,
      lastName: data.user_last_name,
      email: data.user_email,
   },
   hospital: { id: data.hospital_id, name: data.hospital_name },
   ...data.extra_data,
})
