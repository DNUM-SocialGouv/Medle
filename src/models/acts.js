import moment from "moment"

import { ISO_DATE } from "../utils/date"

// from Knex entity to model (JS) entity
export const transform = (knexData) => {
  return !knexData
    ? null
    : {
        id: knexData.id,
        internalNumber: knexData.internal_number,
        pvNumber: knexData.pv_number,
        examinationDate: moment(knexData.examination_date).format(ISO_DATE),
        profile: knexData.profile,
        asker: { id: knexData.asker_id, name: knexData.asker_name },
        user: {
          id: knexData.added_by,
          firstName: knexData.user_first_name,
          lastName: knexData.user_last_name,
          email: knexData.user_email,
        },
        hospital: { id: knexData.hospital_id, name: knexData.hospital_name },
        ...knexData.extra_data,
      }
}
// from Knex entity to model (JS) entity for export needs
export const transformForExport = (knexData) => {
  return !knexData
    ? null
    : {
        id: knexData.id,
        internalNumber: knexData.internal_number,
        pvNumber: knexData.pv_number,
        examinationDate: moment(knexData.examination_date).format(ISO_DATE),
        profile: knexData.profile,
        asker: knexData.asker_name,
        user: knexData.user_email,
        hospital: knexData.hospital_name,
        ...knexData.extra_data,
      }
}

export const transformAll = (list) => list.map((knexData) => transform(knexData))
export const transformAllForExport = (list) => list.map((knexData) => transformForExport(knexData))

// from model (JS) entity to Knex entity
export const untransform = (model) => {
  const knexData = { extra_data: {} }

  const mainKeys = {
    id: "id",
    internalNumber: "internal_number",
    pvNumber: "pv_number",
    examinationDate: "examination_date",
    profile: "profile",
    askerId: "asker_id",
    addedBy: "added_by",
    hospitalId: "hospital_id",
  }

  Object.keys(model).forEach((key) => {
    if (mainKeys[key]) {
      knexData[mainKeys[key]] = model[key]
    } else {
      knexData.extra_data[key] = model[key]
    }
  })

  return knexData
}
