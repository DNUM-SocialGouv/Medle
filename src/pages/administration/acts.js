import { PropTypes } from "prop-types"
import React from "react"
import { Container } from "reactstrap"

import Layout from "../../components/Layout"
import { Title1 } from "../../components/StyledComponents"
import { withAuthentication } from "../../utils/auth"
import { ADMIN } from "../../utils/roles"

const ActsPage = ({ currentUser }) => {
  return (
    <Layout page="acts" currentUser={currentUser} admin={true}>
      <Container
        style={{ maxWidth: 980, minWidth: 740 }}
        className="mt-5 mb-5 d-flex justify-content-between align-items-baseline"
      >
        <Title1 className="">{"Administration des actes"}</Title1>
      </Container>
    </Layout>
  )
}

ActsPage.propTypes = {
  paginatedData: PropTypes.object,
  currentUser: PropTypes.object.isRequired,
}

export default withAuthentication(ActsPage, ADMIN)
