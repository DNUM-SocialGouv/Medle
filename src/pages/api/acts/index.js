import Cors from "micro-cors"

import { STATUS_200_OK, METHOD_GET, METHOD_OPTIONS, METHOD_POST } from "../../../utils/http"
import { ACT_CONSULTATION, ACT_MANAGEMENT } from "../../../utils/roles"
import { sendAPIError, sendMethodNotAllowedError } from "../../../services/errorHelpers"
import { checkValidUserWithPrivilege } from "../../../utils/auth"

import { create, search } from "../../../services/acts"

const handler = async (req, res) => {
   res.setHeader("Content-Type", "application/json")

   try {
      switch (req.method) {
         case METHOD_GET: {
            const currentUser = checkValidUserWithPrivilege(ACT_CONSULTATION, req, res)

            const { totalCount, currentPage: requestedPage, maxPage, byPage: LIMIT, elements: acts } = await search(
               req.query,
               currentUser,
            )

            return res
               .status(STATUS_200_OK)
               .json({ totalCount, currentPage: requestedPage, maxPage, byPage: LIMIT, elements: acts })
         }
         case METHOD_POST: {
            const currentUser = checkValidUserWithPrivilege(ACT_MANAGEMENT, req, res)

            const id = await create(req.body, currentUser)

            return res.status(STATUS_200_OK).json({ id })
         }

         default:
            return sendMethodNotAllowedError(res)
      }
   } catch (error) {
      sendAPIError(error, res)
   }
}

const cors = Cors({
   allowMethods: [METHOD_GET, METHOD_OPTIONS, METHOD_POST],
})

export default cors(handler)
