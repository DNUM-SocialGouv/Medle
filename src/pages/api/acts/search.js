import Cors from "micro-cors"

import { STATUS_200_OK, STATUS_500_INTERNAL_SERVER_ERROR, METHOD_GET, METHOD_OPTIONS } from "../../../utils/http"
import knex from "../../../knex/knex"
import { ACT_CONSULTATION } from "../../../utils/roles"
import { checkValidUserWithPrivilege, sendAPIError } from "../../../utils/api"
import { APIError } from "../../../utils/errors"

const LIMIT = 100

const makeWhereClause = ({ scope, internalNumber, pvNumber, fuzzy }) => builder => {
   builder.whereNull("acts.deleted_at")
   if (scope && scope.length) {
      builder.where(knex.raw("hospital_id in (" + scope.map(() => "?").join(",") + ")", [...scope]))
   }

   if (internalNumber) {
      builder.where("internal_number", internalNumber)
   }

   if (pvNumber) {
      builder.where("pv_number", pvNumber)
   }

   if (fuzzy) {
      builder.where(function() {
         this.where("internal_number", "ilike", `%${fuzzy}%`)
            .orWhere("pv_number", "ilike", `%${fuzzy}%`)
            .orWhere("profile", "ilike", `%${fuzzy}%`)
      })
   }
}

const handler = async (req, res) => {
   res.setHeader("Content-Type", "application/json")

   try {
      // privilege verification
      const currentUser = checkValidUserWithPrivilege(ACT_CONSULTATION, req, res)

      let scope = currentUser.scope || []
      scope = [...scope, currentUser.hospitalId]

      const { fuzzy, internalNumber, pvNumber } = req.query

      let requestedPage =
         req.query.requestedPage && !isNaN(req.query.requestedPage) && parseInt(req.query.requestedPage)

      try {
         const [actsCount] = await knex("acts")
            .where(makeWhereClause({ scope, internalNumber, pvNumber, fuzzy }))
            .count()

         const totalCount = parseInt(actsCount.count)
         const maxPage = Math.ceil(totalCount / LIMIT)

         // set default to 1 if not correct or too little, set default to maxPage if too big
         requestedPage =
            !requestedPage || isNaN(requestedPage) || requestedPage < 1
               ? 1
               : requestedPage > maxPage
               ? maxPage
               : requestedPage

         const offset = (requestedPage - 1) * LIMIT

         // SQL query
         const acts = await knex("acts")
            .where(makeWhereClause({ scope, internalNumber, pvNumber, fuzzy }))
            .orderBy([{ column: "acts.created_at", order: "desc" }])
            .limit(LIMIT)
            .offset(offset)
            .select("*")

         return res.status(STATUS_200_OK).json({ totalCount, currentPage: requestedPage, maxPage, byPage: LIMIT, acts })
      } catch (error) {
         throw new APIError({
            status: STATUS_500_INTERNAL_SERVER_ERROR,
            message: "Erreur DB",
            detailMessage: error.message,
            uri: "https://docs.postgresql.fr/8.3/errcodes-appendix.html",
         })
      }
   } catch (error) {
      // DB error
      sendAPIError(error, res)
   }
}

const cors = Cors({
   allowMethods: [METHOD_GET, METHOD_OPTIONS],
})

export default cors(handler)
