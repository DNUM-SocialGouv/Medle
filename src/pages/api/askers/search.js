import Cors from "micro-cors"

import { STATUS_200_OK, METHOD_GET, METHOD_OPTIONS } from "../../../utils/http"
import knex from "../../../knex/knex"
import { ACT_MANAGEMENT } from "../../../utils/roles"
import { checkValidUserWithPrivilege, sendAPIError } from "../../../utils/api"

const MAX_VALUE = 100000

const handler = async (req, res) => {
   res.setHeader("Content-Type", "application/json")

   try {
      // privilege verification
      checkValidUserWithPrivilege(ACT_MANAGEMENT, req, res)

      const { fuzzy, all } = req.query

      // SQL query
      const askers = await knex("askers")
         .whereNull("deleted_at")
         .where(builder => {
            if (fuzzy) {
               builder.where("name", "ilike", `%${fuzzy}%`)
            }
         })
         .limit(all ? MAX_VALUE : 10)
         .orderBy("name")
         .select("id as value", "name as label")

      return res.status(STATUS_200_OK).json(askers)
   } catch (error) {
      // DB error
      sendAPIError(error, res)
   }
}

const cors = Cors({
   allowMethods: [METHOD_GET, METHOD_OPTIONS],
})

export default cors(handler)
