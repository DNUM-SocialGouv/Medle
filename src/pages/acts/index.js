import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown"
import ArrowRightIcon from "@material-ui/icons/ArrowRight"
import ListAltIcon from "@material-ui/icons/ListAlt"
import Link from "next/link"
import PropTypes from "prop-types"
import React, { useEffect, useMemo, useState } from "react"
import { useForm } from "react-hook-form"
import Select from "react-select"
import AsyncSelect from "react-select/async"
import { Alert, Button, Col, Container, Form, FormGroup, Input, Label, Row, Table } from "reactstrap"

import { fetchExport, searchActs } from "../../clients/acts"
import { memoizedSearchAskers } from "../../clients/askers"
import { SearchButton } from "../../components/form/SearchButton"
import Layout from "../../components/Layout"
import Pagination from "../../components/Pagination"
import { Title1 } from "../../components/StyledComponents"
import { VerticalList } from "../../components/VerticalList"
import { isOpenFeature } from "../../config"
import { useDebounce } from "../../hooks/useDebounce"
import { usePaginatedData } from "../../hooks/usePaginatedData"
import { buildScope } from "../../services/scope"
import { profiles as profilesConstants } from "../../utils/actsConstants"
import { buildAuthHeaders, redirectIfUnauthorized, withAuthentication } from "../../utils/auth"
import { isoToFr } from "../../utils/date"
import { getReferenceData } from "../../utils/init"
import { logError } from "../../utils/logger"
import { ACT_CONSULTATION } from "../../utils/roles"
import { mapArrayForSelect } from "../../utils/select"

const ActsListPage = ({ paginatedData: initialPaginatedData, currentUser }) => {
  // const renderCount = React.useRef(0)
  // renderCount.current++

  const [paginatedData, error, loading, fetchPage] = usePaginatedData(searchActs, initialPaginatedData)
  const [isOpenedFilters, setOpenedFilters] = useState(false)
  const { register, unregister, handleSubmit, setValue, getValues } = useForm({})
  const [hospitals, setHospitals] = useState([])
  const [profiles, setProfiles] = useState([])
  const [asker, setAsker] = useState(null)
  const [search, setSearch] = useState("")
  useDebounce(onChange, 500, [search])
  const scope = useMemo(() => buildScope(currentUser), [currentUser])

  const hospitalsChoices = useMemo(
    () =>
      mapArrayForSelect(
        scope?.length === 0
          ? getReferenceData("hospitals")
          : getReferenceData("hospitals").filter((hospital) => scope.includes(hospital.id)),
        (elt) => elt.id,
        (elt) => elt.name
      ),
    [scope]
  )

  const existingProfiles = useMemo(
    () =>
      mapArrayForSelect(
        Object.keys(profilesConstants) || [],
        (elt) => elt,
        (elt) => elt
      ),
    []
  )

  useEffect(() => {
    // Extra field in form to store the value of selects
    register({ name: "hospitals" })
    register({ name: "profiles" })
    register({ name: "asker" })
    register({ name: "search" })

    return () => {
      unregister({ name: "hospitals" })
      unregister({ name: "profiles" })
      unregister({ name: "asker" })
      unregister({ name: "search" })
    }
  }, [register, unregister])

  const numFilters = Object.values(getValues()).filter((val) => !!val).length

  function toggleFilters() {
    setOpenedFilters((state) => !state)
    onChange()
  }

  function onHospitalsChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("hospitals", !selectedOption?.length ? null : selectedOption)
    // Needs to sync specifically the value to the react-select as well
    setHospitals(selectedOption)
    onChange()
  }

  function onProfilesChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("profiles", !selectedOption?.length ? null : selectedOption)

    // Needs to sync specifically the value to the react-select as well
    setProfiles(selectedOption)
    onChange()
  }

  function onAskerChange(selectedOption) {
    // Needs transformation between format of react-select to expected format for API call
    setValue("asker", selectedOption)

    // Needs to sync specifically the value to the react-select as well
    setAsker(selectedOption)
    onChange()
  }

  function onSearchChange(e) {
    const text = e?.target?.value || ""
    setSearch(text)
    setValue("search", text)
  }

  async function loadAskers(search) {
    const askers = await memoizedSearchAskers({ search })

    return mapArrayForSelect(
      askers?.elements,
      (elt) => elt.id,
      (elt) => elt.name + (elt.depCode ? ` (${elt.depCode})` : "")
    )
  }

  function onChange() {
    onSubmit(getValues())
  }

  function onSubmit(formData) {
    fetchPage(formData)(0)
  }

  async function onExport() {
    await fetchExport(getValues())
  }

  return (
    <Layout page="acts" currentUser={currentUser}>
      <Title1 className="mt-5 mb-4">{"Tous les actes"}</Title1>
      <Container style={{ maxWidth: 980 }}>
        <Form onSubmit={handleSubmit(onSubmit)}>
          <FormGroup inline className="mb-4 justify-content-center">
            <Row>
              <Col className="flex-grow-1">
                <Input
                  type="text"
                  name="search"
                  id="search"
                  placeholder="Rechercher un acte ou plusieurs, par n° interne ou n° de PV"
                  autoComplete="off"
                  value={search}
                  onChange={onSearchChange}
                />
              </Col>
            </Row>
            {isOpenFeature("export") && (
              <>
                <Row className="mt-3">
                  <Col>
                    <Button className="mr-3" color="secondary" outline={!isOpenedFilters} onClick={toggleFilters}>
                      Plus de critères {isOpenedFilters ? <ArrowDropDownIcon /> : <ArrowRightIcon />}
                    </Button>
                    {numFilters > 0 && (
                      <span style={{ color: "#6c757d" }}>
                        {numFilters} critère{numFilters > 1 && "s"} actif{numFilters > 1 && "s"}
                      </span>
                    )}
                  </Col>
                </Row>
                {isOpenFeature("export") && (
                  <div
                    className="p-3 mt-3 border rounded shadow-xl border-light bg-light"
                    style={{ display: isOpenedFilters ? "block" : "none" }}
                  >
                    <Row>
                      <Col sm="3">
                        <Label htmlFor="startDate" className="text-dark">
                          Date de début
                        </Label>
                        <Input
                          type="date"
                          name="startDate"
                          id="startDate"
                          placeholder="Date de début"
                          innerRef={register}
                          onChange={onChange}
                        />
                      </Col>
                      <Col sm="3">
                        <Label htmlFor="startDate" className="text-dark">
                          Date de fin
                        </Label>
                        <Input
                          type="date"
                          name="endDate"
                          id="endDate"
                          placeholder="Date de fin"
                          innerRef={register}
                          onChange={onChange}
                        />
                      </Col>
                    </Row>
                    {hospitalsChoices?.length > 1 && (
                      <Row className="mt-3">
                        <Col>
                          <Label className="text-dark">Établissements</Label>
                          <Select
                            options={hospitalsChoices}
                            value={hospitals}
                            isMulti
                            onChange={onHospitalsChange}
                            noOptionsMessage={() => "Aucun résultat"}
                            placeholder="Choisissez un établissement de votre périmètre"
                            isClearable={true}
                            isSearchable={true}
                          />
                        </Col>
                      </Row>
                    )}{" "}
                    <Row className="mt-3">
                      <Col>
                        <Label className="text-dark">Profils et autres activités</Label>
                        <Select
                          options={existingProfiles}
                          value={profiles}
                          isMulti
                          onChange={onProfilesChange}
                          noOptionsMessage={() => "Aucun résultat"}
                          placeholder="Choisissez un profil ou activité"
                          isClearable={true}
                          isSearchable={true}
                        />
                      </Col>
                    </Row>
                    <Row className="mt-3">
                      <Col>
                        <Label className="text-dark">Demandeur</Label>
                        <AsyncSelect
                          loadOptions={(search) => loadAskers(search)}
                          placeholder="Tapez le nom du demandeur"
                          noOptionsMessage={() => "Aucun résultat"}
                          loadingMessage={() => "Chargement..."}
                          onChange={onAskerChange}
                          isClearable={true}
                          isSearchable={true}
                          value={asker}
                        />
                      </Col>
                    </Row>
                  </div>
                )}
              </>
            )}
          </FormGroup>
        </Form>
        {error && (
          <Alert color="danger" className="mb-4">
            {error}
          </Alert>
        )}
        {!error && !paginatedData?.elements?.length && <div className="text-center">{"Aucun acte trouvé."}</div>}

        {!error && !!paginatedData?.elements?.length && (
          <>
            <div className="my-4 d-flex justify-content-center">
              <b>{paginatedData.totalCount}</b>&nbsp;résultat{paginatedData.totalCount > 1 && "s"}
            </div>
            <Table responsive className="table-hover">
              <thead>
                <tr className="table-light">
                  <th>N° dossier interne</th>
                  <th>N° PV</th>
                  <th>Date</th>
                  <th>Type de profil</th>
                  <th>{"Type d'acte"}</th>
                  <th />
                </tr>
              </thead>
              <tbody>
                {paginatedData.elements.map((act) => (
                  <Link key={act.id} href="/acts/[id]" as={`/acts/${act.id}`}>
                    <tr key={act.id}>
                      <td>
                        <b>{act.internalNumber}</b>
                      </td>
                      <td>{act.pvNumber}</td>
                      <td>{act.examinationDate && isoToFr(act.examinationDate)}</td>
                      <td>{act.profile}</td>
                      <td>{act.examinationTypes && <VerticalList content={act.examinationTypes} />}</td>
                      <td className="text-decoration">
                        <Link href="/acts/[id]" as={`/acts/${act.id}`}>
                          <a className="text-decoration-none">Voir</a>
                        </Link>
                      </td>
                    </tr>
                  </Link>
                ))}
              </tbody>
            </Table>
            <Pagination data={paginatedData} fn={fetchPage(getValues())} />
            {isOpenFeature("export") && (
              <div className="mt-5 d-flex justify-content-center">
                <SearchButton className="btn-outline-primary" disabled={loading} onClick={onExport}>
                  <ListAltIcon /> Exporter les données
                </SearchButton>
              </div>
            )}
          </>
        )}
      </Container>
    </Layout>
  )
}

ActsListPage.getInitialProps = async (ctx) => {
  const headers = buildAuthHeaders(ctx)
  try {
    const paginatedData = await searchActs({ headers })
    return { paginatedData }
  } catch (error) {
    logError("APP error", error)

    redirectIfUnauthorized(error, ctx)
  }
  return {}
}

ActsListPage.propTypes = {
  currentUser: PropTypes.object.isRequired,
  paginatedData: PropTypes.object.isRequired,
}

export default withAuthentication(ActsListPage, ACT_CONSULTATION)
