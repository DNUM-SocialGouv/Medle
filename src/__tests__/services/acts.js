import { normalizeInputs } from "../../services/acts/search"

describe("", () => {
  it("should normalizeInputs", () => {
    const props = {
      scope: [],
      startDate: "2020-05-12",
      endDate: "2020-05-08",
      hospitals: "3,2",
      profiles: "Personne décédée,Personne pour âge osseux (hors GAV)",
      asker: "3849",
      internalNumber: undefined,
      pvNumber: undefined,
      fuzzy: undefined,
      requestedPage: 3,
      currentUser: undefined,
    }

    expect(normalizeInputs(props)).toMatchSnapshot()
  }),
    it("should normalizeInputs with incorrect values", () => {
      const props = {
        scope: [],
        startDate: "15-12",
        endDate: "2020-05-08",
        hospitals: "3,aze, 2",
        profiles: "Personne décédée,Personne pour âge osseux (hors GAV)",
        asker: "3849",
        internalNumber: "123456",
        pvNumber: undefined,
        fuzzy: "Vict",
        requestedPage: -1,
        currentUser: { scope: [1, 3] },
      }

      expect(normalizeInputs(props)).toMatchSnapshot()
    })
})
