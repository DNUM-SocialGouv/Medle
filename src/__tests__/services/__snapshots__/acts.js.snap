// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should normalizeInputs 1`] = `
Object {
  "asker": null,
  "endDate": "2020-05-08",
  "fuzzy": undefined,
  "hospitals": Array [
    3,
    2,
  ],
  "internalNumber": undefined,
  "profiles": Array [
    "Personne décédée",
    "Personne pour âge osseux (hors GAV)",
  ],
  "pvNumber": undefined,
  "requestedPage": 3,
  "scope": Array [],
  "startDate": "2020-05-12",
}
`;

exports[`should normalizeInputs with incorrect values 1`] = `
Object {
  "asker": null,
  "endDate": "2020-05-08",
  "fuzzy": "Vict",
  "hospitals": Array [
    3,
    2,
  ],
  "internalNumber": "123456",
  "profiles": Array [
    "Personne décédée",
    "Personne pour âge osseux (hors GAV)",
  ],
  "pvNumber": undefined,
  "requestedPage": -1,
  "scope": Array [
    1,
    3,
  ],
  "startDate": null,
}
`;
